spark-plugins/src/main/java/io/cdap/plugin/spark/DataCacher.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright © 2016-2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.spark;
+
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.cdap.etl.api.batch.SparkCompute;
+import io.cdap.cdap.etl.api.batch.SparkExecutionPluginContext;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.storage.StorageLevel;
+
+/**
+ * SparkCompute that caches a RDD
+ */
+@Plugin(type = SparkCompute.PLUGIN_TYPE)
+@Name("DataCacher")
+@Description("Spark Data Cacher caches any incoming records and outputs them unchanged.")
+public class DataCacher extends SparkCompute<StructuredRecord, StructuredRecord> {
+
+  private final DataCacherConfig config;
+
+  public DataCacher(DataCacherConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public JavaRDD<StructuredRecord> transform(
+    SparkExecutionPluginContext sparkExecutionPluginContext, JavaRDD<StructuredRecord> javaRDD)
+    throws Exception {
+
+    StorageLevel storageLevel = config.getStorageLevel();
+    if (storageLevel == StorageLevel.NONE()) {
+      throw new RuntimeException(String.format("Invalid storage level '%s'. Please select a valid value"));
+    }
+
+    javaRDD.persist(storageLevel);
+    return javaRDD;
+  }
+
+  /**
+   * Config class for DataCacher.
+   */
+  public static class DataCacherConfig extends PluginConfig {
+
+    @Name("StorageLevel")
+    @Description("Spark storage level used to cache the data")
+    @Macro
+    private String storageLevel;
+
+    StorageLevel getStorageLevel() {
+      if (Strings.isNullOrEmpty(storageLevel)) {
```

Comment:
You can just check for empty string since the config is not optional (i.e. doesn't have `@Nullable` annotation)

Reply:
Fixed